src/server/project.ts

```diff
@@ -1815,7 +1815,7 @@ namespace ts.server {
                     name,
                     rootFileName,
                     compilerOptions,
-                    hostProject));
+                    moduleResolutionHost));

```

Comment:
You dont need to pass moduleResolutionHost here but instead you want this to also handle trace (if logging enabled going to log) and currentDIrectory
ModuleResolutionHost = hostProject.projectService ?

Reply:
I don’t think I understand the suggestion—`ProjectService` isn’t assignable to `ModuleResolutionHost`.

Reply:
Merging for the beta, feel free to follow up with me

Reply:
You want it to be {
boundMethods from hostProject.projectService.host
getCurrentDirectory : () => hostProject.getCurrentDirectory(),
trace: mayBeBind(hostProject, hostProject.trace)
}

Reply:
Is there a realistic example of why `hostProject` itself wouldn’t work? If it wants to mimic the existence of some files, why shouldn’t it be allowed to do that during type directive resolution?